hikari/impl/rest.py

```diff
@@ -2095,9 +2095,20 @@ async def fetch_ban(
         self,
         guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
         user: snowflakes.SnowflakeishOr[users.PartialUser],
-    ) -> guilds.GuildMemberBan:
+    ) -> typing.Optional[guilds.GuildMemberBan]:
         route = routes.GET_GUILD_BAN.compile(guild=guild, user=user)
-        raw_response = await self._request(route)
+        try:
+            raw_response = await self._request(route)
+        except errors.NotFoundError as ex:
+            # Discord raises an error when the ban is not found.
+            # We simplify this for the user and only raise the
+            # NotFoundError if it is something else not being found.
+            json_body = typing.cast(data_binding.JSONObject, data_binding.load_json(ex.raw_body))
+            if json_body["code"] == 10026:
```

Comment:
We should define an enum with this code in if we add any more in the future.

Reply:
Could do, but since this is the only one where this happens. One thing I thought about was maybe having a `code` property in the errors that will return the code. Will have a think 👍

Reply:
Special cases arent special enough to break the rules, right?